ipc/provider/src/manager/evm/manager.rs

```diff
@@ -333,6 +333,9 @@ impl SubnetManager for EthSubnetManager {
         txn.tx.set_value(collateral);
         let txn = call_with_premium_estimation(signer, txn).await?;
 
+        // Use the pending state to get the nonce because there could have been a pre-fund. Best would be to use this for everything.
+        let txn = txn.block(BlockId::Number(ethers::types::BlockNumber::Pending));
+
```

Comment:
I don't know. I didn't want everything to always necessarily use `Pending` because it's a bottleneck, OTOH invoking the `ipc-cli` doesn't happen too frequently. I'd leave it up to @cryptoAtwill judgement where it should be. Ostensibly we could even use an ethers layer that maintains the nonce on the client side, so it doesn't get queried again after it has just submitted a transaction.